components/playlist/browser/resources/components/playlistItem.tsx

```diff
@@ -69,9 +69,30 @@ export default function PlaylistItem ({
   cached,
   onClick
 }: Props) {
+  const anchorElem = React.useRef<HTMLAnchorElement>(null)
+
+  const scrollToThisIfNeeded = () => {
+    if (window.location.hash.replace('#', '') !== id) return
+
+    if (anchorElem.current)
+      window.scrollTo({ top: anchorElem.current.offsetTop })
+  }
+
+  React.useEffect(() => {
+    scrollToThisIfNeeded()
```

Comment:
So this is *probably* fine unless the `id` for this component changes (say, if we were rendering this in a list and not keying by `id`).

`scrollToThisIfNeeded` is captured at the time the effect runs (which is once, at component mount for this effect), so it will always have the `id` of the it was first rendered with). Effects in React have a lot of footguns :laughing:

(if you want to go down the rabbit hole, [this issue thread](https://github.com/facebook/react/issues/14099#issuecomment-1407393353) proposed a `useEvent` hook which would nicely solve this for you, but it's kinda in limbo). (the RFC + discussion is here https://github.com/reactjs/rfcs/pull/220)

But, if the id is expected to be stable, I think this is okay - maybe to make it more obvious what's going on you could move `scrollToThisIfNeeded` into the `useEffect` statement (as that's equivalent to how this works now).

Sorry for the minor essay :laughing:

Reply:
Yup, that should solve it nicely :smile: - might break scrolling up and down as items are loaded/unloaded (because they try and target themselves on mount), so maybe something like this:

```tsx

const scrollToThisIfNeeded = useCallback(() => {
    ...
}, [id])

useEffect(() => {
    // add listeners
    return () => { /* remove listeners */ }
}, [scrollToThisIfNeeded])

// onMount effect, only once per component
useEffect(() => scrollToThisIfNeeded(), [])
```

Reply:
It'd be way easier with `useEvent`

```tsx
const scrollToThisIfNeeded = useEvent(() => { ... })
useEffect(() => {
    // the existing implementation
}, [])
```

Reply:
I really appreciate your explanation. Learned a lot! 

useEvent is fascinating that it always references the latest props/states! It's too bad it's still in experimental stage.